CRM/Contribute/Tokens.php

```diff
@@ -194,7 +194,43 @@ public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL)
       $row->tokens($entity, $field, $fieldValue);
     }
     else {
-      $row->dbToken($entity, $field, 'CRM_Contribute_BAO_Contribution', $field, $fieldValue);
+      // This section is copied & pasted & can be cleaned up later.
+      // most if probably already done in this functions
+      $tokenEntity = $entity;
+      $tokenField = $field;
+      $baoName = $this->getBAOName();
+      $baoField = $field;
+      if ($fieldValue === NULL || $fieldValue === '') {
+        return $row->tokens($tokenEntity, $tokenField, '');
+      }
+
+      $fields = $this->getFieldMetadata();
```

Comment:
@seamuslee001 note that the change required to get this PR to work is `$fields = $this->getFieldMetadata();` - because otherwise the metadata is keyed by our legacy fields in the dbToken function.